src/components/Login/Login.js

```diff
@@ -1,26 +1,66 @@
-import React from 'react';
+import React, { useEffect, useRef } from 'react';
+import fetch from 'isomorphic-fetch';
 import styles from  './Login.css';
 import useStoreon from 'storeon/react'
 import Header from '../common/Header';
 
 const Login = () => {
     const { user, dispatch } = useStoreon('user');
+    const usernameEl = useRef(null);
+    const passwordEl = useRef(null);
+    
+    useEffect(()=>{
+        console.log('useEffect');
+        console.log('user', user);
+    });
+
     return(
         <div className={styles['login-page']}>
             <Header title={'View service details'} />
             <div className={styles["form"]}>
                 <div className={styles['input-row']}>
-                    <input type="text" name="username" placeholder="Username" />
+                    <input ref={usernameEl} type="text" name="username" placeholder="Username" />
                 </div>
                 <div className={styles['input-row']}>
-                    <input type="password" name="password" placeholder="Password" />
+                    <input ref={passwordEl} type="password" name="password" placeholder="Password" />
                 </div>
                 <div className={styles['input-row']}>
-                    <button type="button" onClick={ ()=> dispatch('user:loggedin') }>Login</button>
+                    <button type="button" onClick={onLogin.bind(null, dispatch, usernameEl, passwordEl)}>Login</button>
```

Comment:
bind with out context is used here. Bind will return a new function. So I think the code will be equivalent to onClick={ ()=>onLogin(dispatch, usernameEl, passwordEl)}. Please check.

Reply:
Good point. No need to bind. Moved to the code you suggested.

But, it still returns a new function. There's no way of escaping it I guess.